### PR TITLE
feat: Supports new `networking` attribute in `mongodbatlas_stream_connection`

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Supports new `networking` attribute in `mongodbatlas_stream_connection`
This attribute is only used when type of connection is Kafka.
Attribute is Computed and Optional. This is because it can be set by the user but it also has a default value if is not set by the user and is always returned by the API.

Link to any related issue(s): CLOUDP-265074

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/streamconnection/data_source_stream_connection_test.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func TestAccStreamDSStreamConnection_kafkaPlaintext(t *testing.T) {
 		CheckDestroy:             CheckDestroyStreamConnection,
 		Steps: []resource.TestStep{
 			{
-				Config: streamConnectionDataSourceConfig(kafkaStreamConnectionConfig(projectID, instanceName, "user", "rawpassword", "localhost:9092,localhost:9092", "earliest", false)),
+				Config: streamConnectionDataSourceConfig(kafkaStreamConnectionConfig(projectID, instanceName, "user", "rawpassword", "localhost:9092,localhost:9092", "earliest", false, true)),
```
Comment (EspenAlbert) [inline internal/service/streamconnection/data_source_stream_connection_test.go]: Can we move these tests to the resource tests?

File: internal/service/streamconnection/resource_stream_connection_migration_test.go [hunk 2]
```diff
@@ -24,12 +24,12 @@ func TestMigStreamRSStreamConnection_kafkaPlaintext(t *testing.T) {
 		Steps: []resource.TestStep{
 			{
 				ExternalProviders: mig.ExternalProviders(),
-				Config:            kafkaStreamConnectionConfig(projectID, instanceName, "user", "rawpassword", "localhost:9092,localhost:9092", "earliest", false),
+				Config:            kafkaStreamConnectionConfig(projectID, instanceName, "user", "rawpassword", "localhost:9092,localhost:9092", "earliest", false, false),
```
Comment (EspenAlbert) [inline internal/service/streamconnection/resource_stream_connection_migration_test.go]: Can we refactor these to use the `mig.CreateAndRunTest`?

**Review by EspenAlbert — APPROVED** [review]
LGTM

File: .changelog/2474.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+```release-note:enhancement
+resource/mongodbatlas_stream_connection: Supports new networking attribute
```
Comment (AgustinBettati) [inline .changelog/2474.txt]: nit: would define attribute with backticks and use the full path

File: internal/service/streamconnection/model_stream_connection_test.go [hunk 4]
```diff
@@ -212,6 +218,11 @@ func TestStreamConnectionsSDKToTFModel(t *testing.T) {
 							Protocol:                admin.PtrString(securityProtocol),
 							BrokerPublicCertificate: admin.PtrString(DummyCACert),
 						},
+						Networking: &admin.StreamsKafkaNetworking{
+							Access: &admin.StreamsKafkaNetworkingAccess{
+								Type: admin.PtrString(networkingType),
```
Comment (AgustinBettati) [inline internal/service/streamconnection/model_stream_connection_test.go:223]: could we have a unit test that also checks correct mapping of the `access.name` property? mainly since it is not defined in acceptance tests either.

**Review by AgustinBettati — COMMENTED** [review]
leaving suggestions but in general LGTM

File: .changelog/2474.txt (hunk 3 above)
Comment (oarbusi) [reply .changelog/2474.txt]: What do you mean by full path?

File: internal/service/streamconnection/resource_stream_connection_migration_test.go (hunk 2 above)
Comment (oarbusi) [reply internal/service/streamconnection/resource_stream_connection_migration_test.go]: I don't think it makes sense to refactor each migration test to use `mig.CreateAndRunTest` because Config of step 1 and step 2 changes (1 step does not have `networking`, second step does), and from the code I see: `// CreateTest returns a new TestCase that reuses step 1 and adds a TestStepCheckEmptyPlan.`

File: .changelog/2474.txt (hunk 3 above)
Comment (AgustinBettati) [reply .changelog/2474.txt]: my bad, thought networking was nested within existing attribute but I see it is root level

File: internal/service/streamconnection/model_stream_connection_test.go (hunk 4 above)
Comment (oarbusi) [reply internal/service/streamconnection/model_stream_connection_test.go:223]: Included `name` into the unit tests. Thanks

File: .changelog/2474.txt (hunk 3 above)
Comment (oarbusi) [reply .changelog/2474.txt]: Okay, I have added the backticks to the attribute

File: internal/service/streamconnection/resource_stream_connection_migration_test.go (hunk 2 above)
Comment (AgustinBettati) [reply internal/service/streamconnection/resource_stream_connection_migration_test.go]: would question if we need to test the update as part of the migration test, checking an empty plan after adopting latest version is the regular case of migration tests.

**Review by AgustinBettati — APPROVED** [review]
LGTM

**Review by corryroot — APPROVED** [review]
LGTM!

**Comment by oarbusi** [comment]
Waiting to merge this until stream_connection is not created by default when creating a stream_instance

File: internal/service/streamconnection/resource_stream_connection_migration_test.go (hunk 2 above)
Comment (AgustinBettati) [reply internal/service/streamconnection/resource_stream_connection_migration_test.go]: wanted to follow up here to confirm if an empty plan is encountered when upgrading to latest version, or do users need to add `networking` once they upgrade to obtain an empty plan? @EspenAlbert I see you included plan modifiers but not certain if it is intended for covering this case.

Comment (oarbusi) [reply internal/service/streamconnection/resource_stream_connection_migration_test.go]: My understanding is that those changes avoid plan changes if user does not add the `networking` attribute once they upgrade. But please @EspenAlbert confirm, also might be necessary to change the migration tests now that the plan does not change?

Comment (EspenAlbert) [reply internal/service/streamconnection/resource_stream_connection_migration_test.go]: Yes, it leads to no plan changes. Will push updates when we are unblocked

File: internal/service/streamconnection/resource_stream_connection_test.go [hunk 5]
```diff
@@ -65,6 +66,10 @@ func testCaseKafkaPlaintext(t *testing.T) *resource.TestCase {
 		},
 	}
 }
+func DummySleep(*terraform.State) error {
```
Comment (EspenAlbert) [inline internal/service/streamconnection/resource_stream_connection_test.go]: Merge blocker: Getting 500s when a stream connection changes networking to VPC and is deleted straight afterwards. [Detailed Jira Ticket](https://jira.mongodb.org/browse/CLOUDP-268853)

**Comment by EspenAlbert** [comment]
Merge blocked by [upstream breaking change](https://jira.mongodb.org/browse/CLOUDP-268660), wait for this and remove `access.name` attribute

File: .changelog/2474.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+```release-note:enhancement
+resource/mongodbatlas_stream_connection: Supports new `networking` attribute
```
Comment (lantoli) [inline .changelog/2474.txt]:
```suggestion
resource/mongodbatlas_stream_connection: Adds `networking` attribute
```

nit for all changelog entries

File: internal/service/streamconnection/resource_stream_connection_test.go [hunk 7]
```diff
@@ -13,24 +14,44 @@ import (
 
 //go:embed testdata/dummy-ca.pem
 var DummyCACert string
+var (
```
Comment (lantoli) [inline internal/service/streamconnection/resource_stream_connection_test.go:15]: nit: add DummyCACert to the var block

**Review by lantoli — APPROVED** [review]

File: internal/service/streamconnection/resource_stream_connection_test.go (hunk 7 above)
Comment (EspenAlbert) [reply internal/service/streamconnection/resource_stream_connection_test.go:15]: changed in 6f35f2a4e9b54f4d296ad6e55b085b767039861a

File: .changelog/2474.txt (hunk 6 above)
Comment (EspenAlbert) [reply .changelog/2474.txt]: changed in 0f965c50b5d71792eb9e730a74600c2078f21b10

File: internal/service/streamconnection/resource_stream_connection_test.go (hunk 5 above)
Comment (EspenAlbert) [reply internal/service/streamconnection/resource_stream_connection_test.go]: removed in 86e0e357d9ce9399394228edea56023fc88269e4

**Review by EspenAlbert — APPROVED** [review]

**Comment by oarbusi** [comment]
LGTM, but can we make sure latest [API changes](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Streams/operation/createStreamConnection) are included in this PR? (connectionId, and the fact that name is not needed for VPC)

**Review by andreaangiolillo — APPROVED** [review]

**Comment by EspenAlbert** [comment]
Only failing test is: [TestAccStreamProcessor_withOptions](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/12465704333/job/34792188099?pr=2474#step:5:205) which also fails in master.
